javaslang/src/test/java/javaslang/collection/IteratorTest.java

```diff
@@ -405,6 +401,30 @@ public void shouldNonNilGroupByEqual() {
         assertThat(actual).isEqualTo(expected);
     }
 
+    @Test
+    public void shouldCreateDoubleRangeByFromInfinity() {
+        assertThat(rangeBy(Double.NEGATIVE_INFINITY, 0.0, 1.0)).startsWith(Double.NEGATIVE_INFINITY, Math.nextUp(Double.NEGATIVE_INFINITY));
+        assertThat(rangeBy(Double.POSITIVE_INFINITY, 0.0, -1.0)).startsWith(Double.POSITIVE_INFINITY, Math.nextDown(Double.POSITIVE_INFINITY));
+    }
+
+    @Test
+    public void shouldCreateDoubleRangeClosedByFromInfinity() {
+        assertThat(rangeClosedBy(Double.NEGATIVE_INFINITY, 0.0, 1.0)).startsWith(Double.NEGATIVE_INFINITY, Math.nextUp(Double.NEGATIVE_INFINITY));
+        assertThat(rangeClosedBy(Double.POSITIVE_INFINITY, 0.0, -1.0)).startsWith(Double.POSITIVE_INFINITY, Math.nextDown(Double.POSITIVE_INFINITY));
+    }
+
+    @Test
+    public void shouldCreateDoubleRangeByFromMaxToInfinity() {
+        assertThat(rangeBy(Double.MAX_VALUE, Double.POSITIVE_INFINITY, 3E307)).isEqualTo(of(Double.MAX_VALUE));
+        assertThat(rangeBy(-Double.MAX_VALUE, Double.NEGATIVE_INFINITY, -3E307)).isEqualTo(of(-Double.MAX_VALUE));
+    }
+
+    @Test
+    public void shouldCreateDoubleRangeClosedByFromMaxToInfinity() {
+        assertThat(rangeClosedBy(Double.MAX_VALUE, Double.POSITIVE_INFINITY, 3E307)).isEqualTo(of(Double.MAX_VALUE));
```

Comment:
I like that you shortened/simplified the range code. But I'm not sure about the changes regarding included and excluded INFINITY because of ULP. Also here I would take a look how Scala behaves.